bin/tests/cli.rs

```diff
@@ -384,6 +385,60 @@ fn test_exclusion_rules() {
     agent_handle.kill().expect("Could not kill process");
 }
 
+#[tokio::test]
+#[cfg_attr(not(feature = "integration_tests"), ignore)]
+async fn test_metrics_endpoint() {
+    let dir = tempdir().expect("Could not create temp dir").into_path();
+    let included_file = dir.join("file1.log");
+    let port = 9881;
+
+    let mut agent_handle = common::spawn_agent(AgentSettings {
+        log_dirs: &dir.to_str().unwrap(),
+        metrics_port: Some(port),
+        ..Default::default()
+    });
+
+    let mut stderr_reader = BufReader::new(agent_handle.stderr.take().unwrap());
+    common::wait_for_event("Enabling prometheus endpoint", &mut stderr_reader);
+
+    // Append to a file and wait for the notify event
+    common::append_to_file(&included_file, 100, 5).unwrap();
+    common::wait_for_file_event("tailer sendings lines", &included_file, &mut stderr_reader);
+    let mut body_str = String::new();
+
+    // Wait for all the metrics to be tracked
+    for _ in 0..20 {
+        let client = Client::new();
+        let url = format!("http://127.0.0.1:{}/metrics", port)
+            .parse()
+            .unwrap();
+        if let Ok(response) = client.get(url).await {
```

Comment:
Should this just unwrap? it looks like it will skip the results if client.get fails.

Reply:
yeah, that's expected, it's a way to wait for the endpoint to be ready (I've initially got failures because it was taking ~200ms for endpoint to accept connections).

If it fails several times, the test will fail (`body_str` will be empty).